src/ts/Step.class.ts

```diff
@@ -39,11 +39,15 @@ export default class Step {
   }
 
   finish(id: number): Promise<Response> {
-    return this.api.patch(`${this.entity.type}/${this.entity.id}/${this.model}/${id}`, {'action': 'finish'});
+    return this.genericPatch(id, Action.Finish);
   }
 
   notif(id: number): Promise<Response> {
-    return this.api.patch(`${this.entity.type}/${this.entity.id}/${this.model}/${id}`, {'action': 'notif'});
+    return this.genericPatch(id, Action.Notif);
+  }
+
+  genericPatch(id: number, action: Action): Promise<Response> {
+    return this.api.patch(`${this.entity.type}/${this.entity.id}/${this.model}/${id}`, {action});
```

Comment:
shouldn't it be `{'action' : action}` here? For some reason it seems to work nonetheless! (and I don't really understand why)

Reply:
> When compiled to JavaScript, an enum will be transformed into an object where keys are the names of the fields and the values are the corresponding enum values

ChatGPT, with a screenshot of this part of the diff!